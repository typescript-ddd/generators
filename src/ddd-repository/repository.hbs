import { Repository } from "@typescript-ddd/core";
import { {{pascal entityName}}, {{pascal entityName}}Id } from "../../domain/models";

/**
 * Represents the interface for {{noun entityName}} repository.
 */
export interface {{pascal entityName}}Repository extends Repository<{{pascal entityName}}Id, {{pascal entityName}}> {}