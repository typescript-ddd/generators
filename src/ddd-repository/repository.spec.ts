import { RepositoryTemplateData } from "./repository.types";
import { generateRepository } from "./repository.generator";

describe("Repository", () => {
  it("should render template", () => {
    // Arrange
    const data: RepositoryTemplateData = {
      entityName: "User"
    };

    // Act
    const result = generateRepository(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
