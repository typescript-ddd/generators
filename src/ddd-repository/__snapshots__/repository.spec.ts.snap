// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository should render template 1`] = `
"import { Repository } from "@typescript-ddd/core";
import { User, UserId } from "../../domain/models";

/**
 * Represents the interface for a user repository.
 */
export interface UserRepository extends Repository<UserId, User> {}"
`;
